File: Axiom/Internal/Ast/FunctionExpression.cs
using System.Collections.Generic;

namespace Axiom.Internal.Ast
{
    internal class FunctionExpression : Expression
    {
        public FunctionExpression(IList<Identifier> parameters, Statement body)
        {
            Parameters = parameters;
            Body = body;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public IList<Identifier> Parameters { get; private set; }
        public Statement Body { get; set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/FunctionExpression.java
package com.moybl.axiom.ast;

import com.moybl.axiom.Position;

import java.util.List;

public class FunctionExpression extends Expression {

	private List<Identifier> parameters;
	private Statement body;

	public FunctionExpression(Position position, List<Identifier> parameters, Statement body) {
		super(position);
		this.parameters = parameters;
		this.body = body;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public List<Identifier> getParameters() {
		return parameters;
	}

	public Statement getBody() {
		return body;
	}

}
